sympy/core/expr.py

```diff
@@ -146,11 +146,15 @@ def _hashable_content(self):
         return self._args
 
     def __eq__(self, other):
-        try:
-            other = _sympify(other)
-            if not isinstance(other, Expr):
+        if not isinstance(other, Basic):
```

Comment:
Note that `Tuple` defines its own custom `__eq__`: https://github.com/sympy/sympy/blob/ed0bfd1e47345c8bd45ef542eef18f6a21c55685/sympy/core/containers.py#L96-L99

Although I do believe this is handled correctly.